tests/ops/functions/test_matrix.py

```diff
@@ -191,6 +191,94 @@ def testcircuit():
         assert np.allclose(matrix, expected_matrix)
 
 
+class TestWithParameterBroadcasting:
+    def test_multiple_operations_tape_single_broadcasted_op(self):
+        """Check the total matrix for a tape containing multiple gates
+        and a single broadcasted gate."""
+        wire_order = ["a", "b", "c"]
+
+        angles = np.array([0.0, np.pi, 0.7])
+        with qml.tape.QuantumTape() as tape:
+            qml.S(wires="b")
+            qml.RX(angles, wires="a")
+            qml.Hadamard(wires="c")
+            qml.CNOT(wires=["b", "c"])
+
+        matrix = qml.matrix(tape, wire_order)
+        expected_matrix = [
+            np.kron(I, CNOT) @ np.kron(I, np.kron(S, H)),
+            -1j * np.kron(I, CNOT) @ np.kron(X, np.kron(S, H)),
+            np.kron(I, CNOT) @ np.kron(RX(0.7), np.kron(S, H)),
+        ]

```

Comment:
I think it is fine for this instance (overall runtime is around 1ms), but explicitly creating the matrix here rather than kronning operations would be much more efficient. Even precomputing `np.kron(I, CNOT)` and `np.kron(S, H)` would reduce the number of ops needed here and locally cuts the time in half for me.

In this case , I'd suggest

```suggestion
        I_CNOT = np.kron(I, CNOT)
        S_H = np.kron(S, H)
        expected_matrix = [
            I_CNOT @ np.kron(I, S_H),
            -1j * I_CNOT @ np.kron(X, S_H),
            I_CNOT @ np.kron(RX(0.7), S_H),
        ]
```

Reply:
Oh, I did not look into optimization of this at all :) 
I'll implement these pre-computations for the file, so the kronecker products are only computed once overall.